### PR TITLE
ETH-tests: vm engine

### PR DESCRIPTION
**Pull request by mrLSD**
Related to issue: #170 

Basic source tests for SpuntikVM: [evm-tests](https://github.com/rust-blockchain/evm-tests)

Cover only: VM tests

### CODE REVIEW

File: src/tests/eth/utils.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
+use evm::backend::MemoryAccount;
+use primitive_types::{H160, H256, U256};
+use sha3::{Digest, Keccak256};
+use std::collections::BTreeMap;
+
+pub fn u256_to_h256(u: U256) -> H256 {
```
Comment (birchmd) [inline src/tests/eth/utils.rs:6]: I don't like that we are verbatim copying so [many lines of code](https://github.com/rust-blockchain/evm-tests/blob/master/jsontests/src/utils.rs#L6-L134) from [`evm-jsontests`](https://github.com/rust-blockchain/evm-tests/tree/master/jsontests). Could we add that crate as a dependency instead maybe?

Comment (mrLSD) [reply src/tests/eth/utils.rs:6]: We cant' These crate has bugs. For example, when I tried to run tests from that git repo I received errors.

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "src/tests/eth/ethtests"]
+	path = src/tests/eth/ethtests
+	url = https://github.com/ethereum/tests/
```
Comment (mfornet) [inline .gitmodules:3]: should we lock a particular commit?

Comment (joshuajbouw) [reply .gitmodules:3]: Yes absolutely.

File: src/tests/eth/utils.rs (hunk 1 above)
Comment (joshuajbouw) [reply src/tests/eth/utils.rs:6]: Compatibility errors or?

Comment (joshuajbouw) [inline src/tests/eth/utils.rs:6]: Just a note, I swear that there is an `Into` trait created for `U256.into()` for `H256`..?

Comment (mrLSD) [reply src/tests/eth/utils.rs:6]: Code just can't compile. I mean upstream. So as cate dependency I have the same problem.

Comment (mrLSD) [reply src/tests/eth/utils.rs:6]: utils file just copy-pusted from upstream...

File: .gitmodules (hunk 2 above)
Comment (mrLSD) [reply .gitmodules:3]: Yes, locked particular commit, as in reference implementation. For some reason, VMTests was deleted from https://github.com/ethereum/tests/

File: src/tests/eth/utils.rs (hunk 1 above)
Comment (joshuajbouw) [reply src/tests/eth/utils.rs:6]: Can you check? I don't want to add to tech debt by adding in unnecessary methods if we do not need them.

**Review by mrLSD — COMMENTED** [review]
U256 don't contain `into` implementation for H160

File: src/tests/eth/utils.rs (hunk 1 above)
Comment (mrLSD) [reply src/tests/eth/utils.rs:6]: U256 doesn't contain into implementation for H160

**Review by joshuajbouw — APPROVED** [review]
This is good to go for me then. Just resolve the conflict.

**Comment by joshuajbouw** [comment]
These unfortunately can not be merged into this library due to conflicting license from sourced code. Must target a new library entirely. `aurora-engine-tests` will be fine.

**Pull request closed without merging**